services/viva-ms/lambdas/createVivaCase.js

```diff
@@ -59,58 +59,65 @@ export async function main(event) {
     return console.error('(Viva-ms) Viva Application WorkflowId not present in response, aborting');
   }
 
-  const PK = `USER#${user.personalNumber}`;
-  const [queryCasesError, queryCaseItems] = await to(
-    queryCasesWithWorkflowId(PK, vivaPerson.application.workflowid)
+  const [getUserCaseFilteredOnWorkflowIdError, caseItem] = await to(
+    getUserCaseFilteredOnWorkflowId(vivaPerson)
   );
-  if (queryCasesError) {
-    return console.error('(Viva-ms) DynamoDb query on cases tabel failed', queryCasesError);
+  if (getUserCaseFilteredOnWorkflowIdError) {
+    return console.error(
+      '(Viva-ms) DynamoDb query on cases table failed',
+      getUserCaseFilteredOnWorkflowIdError
+    );
   }
 
-  if (queryCaseItems.length > 0) {
+  if (caseItem.length > 0) {
     return console.log('(Viva-ms) Case with WorkflowId already exists');
   }
 
-  const period = {
-    startDate: Date.parse(vivaPerson.application.period.start),
-    endDate: Date.parse(vivaPerson.application.period.end),
-  };
-
-  const [putItemError] = await to(
-    putRecurringVivaCase(PK, vivaPerson.application.workflowid, period)
-  );
-  if (putItemError) {
-    return console.error('(viva-ms) syncApplicationStatus', putItemError);
+  const [putRecurringVivaCaseError] = await to(putRecurringVivaCase(vivaPerson));
+  if (putRecurringVivaCaseError) {
+    return console.error('(viva-ms) putRecurringVivaCaseError', putRecurringVivaCaseError);
   }
 
   return true;
 }
 
-async function queryCasesWithWorkflowId(PK, workflowId) {
+async function getUserCaseFilteredOnWorkflowId(vivaPerson) {
+  const personalNumber = String(vivaPerson.case.client.pnumber).replace(/\D/g, '');
+  const workflowId = vivaPerson.application.workflowid;
+
   const params = {
     TableName: config.cases.tableName,
     KeyConditionExpression: 'PK = :pk',
     FilterExpression: 'details.workflowId = :workflowId',
     ExpressionAttributeValues: {
-      ':pk': PK,
+      ':pk': `USER#${personalNumber}`,
       ':workflowId': workflowId,
     },
+    Limit: 1,
   };
 
-  const [dynamoQueryError, queryCasesResult] = await to(dynamoDB.call('query', params));
-  if (dynamoQueryError) {
-    throw dynamoQueryError;
+  const [queryCasesError, queryCasesResult] = await to(dynamoDB.call('query', params));
+  if (queryCasesError) {
+    throw queryCasesError;
   }
 
   return queryCasesResult.Items;
```

Comment:
Since we're only fetching one case we should not return a list. 

Suggestion: change to `return queryCasesResult.Items[0] || null`.